### PR TITLE
Add dictionary settings

### LINKED ISSUE #489: [v1.4] Support user-dictionary loading
*Following this [central issue](https://github.com/meilisearch/integration-guides/issues/286)*

- [ ]  Add a new setting: `dictionary` with `get`, `update`, and `reset` methods associated.
Here are the JS equivalents in meilisearch-js you should create for this repository:

```jsx
client.index('indexName').getDictionary(); // calls GET /indexes/:uid/settings/dictionary
client.index('indexName').updateDictionary(['|', '/', '&sep']);  // calls PUT /indexes/:uid/settings/dictionary
client.index('indexName').resetDictionary(); // calls DELETE /indexes/:uid/settings/dictionary
```

- [ ]  The methods associated with the `/settings` API route (to get, reset and update the settings globally) must be able to receive in the payload the new configuration `dictionary`.
- [ ]  ⚠️ Add integration tests ⚠️
- [ ]  Update the `.code-samples.meilisearch.yaml`
    - [ ]  Add `get_dictionary_1` key and “translate” the following curl example by using the newly added methods: https://github.com/meilisearch/documentation/blob/2060089a5c10bcd0beb28633062f90cf0b279f7f/.code-samples.meilisearch.yaml#L1133
    - [ ]  Add `update_dictionary_1` key and “translate” the following curl example by using the newly added methods: https://github.com/meilisearch/documentation/blob/2060089a5c10bcd0beb28633062f90cf0b279f7f/.code-samples.meilisearch.yaml#L1136
    - [ ]  Add `reset_dictionary_1` key and “translate” the following curl example by using the newly added methods: https://github.com/meilisearch/documentation/blob/2060089a5c10bcd0beb28633062f90cf0b279f7f/.code-samples.meilisearch.yaml#L1144

### PR DESCRIPTION
**Pull request by Ja7ad**
# Pull Request

## Related issue
Fixes #489

## What does this PR do?
- This PR Support user-dictionary loading

## PR checklist
Please check if your PR fulfills the following requirements:
- [x] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

**Comment by Ja7ad** [comment]
> @Ja7ad can you fix the git conflict so that I can review properly?

I fixed it

File: .code-samples.meilisearch.yaml [hunk 1]
```diff
@@ -897,8 +897,17 @@ update_search_cutoff_1: |-
...
   client.Index("articles").GetSeparatorTokens()
+get_separator_tokens_1: |-
+  client.Index("books").GetDictionary()
+update_dictionary_1: |-
+  client.Index("books").UpdateSeparatorTokens([]string{
+    "J. R. R.",
+    "W. E. B.",
+  })
+reset_dictionary_1: |-
+  client.Index("books").ResetDictionary()
```
[the remark this answers is not in the document]
Comment (Ja7ad) [reply .code-samples.meilisearch.yaml:910]: > There is confusion here with Dictionary and separator token 😇

Which line have issue?

[the remark this answers is not in the document]
Comment (curquiza) [reply .code-samples.meilisearch.yaml:910]: ensure the code sample named "dictiionnary" is actually updating dictionary and not separator

File: .code-samples.meilisearch.yaml [hunk 2]
```diff
@@ -897,8 +897,17 @@ update_search_cutoff_1: |-
   client.Index("movies").UpdateSearchCutoffMs(150)
 reset_search_cutoff_1: |-
   client.Index("books").ResetSearchCutoffMs()
+get_dictionary_1: |-
+  client.Index("articles").GetDictionary()
```
Comment (curquiza) [inline .code-samples.meilisearch.yaml:901]:
```suggestion
  client.Index("books").GetDictionary()
```

File: .code-samples.meilisearch.yaml [hunk 3]
```diff
@@ -897,8 +897,17 @@ update_search_cutoff_1: |-
...
+update_dictionary_1: |-
+  client.Index("books").UpdateSeparatorTokens([]string{
```
Comment (curquiza) [inline .code-samples.meilisearch.yaml:903]:
```suggestion
  client.Index("books").UpdateDictionary([]string{
```

File: .code-samples.meilisearch.yaml [hunk 4]
```diff
@@ -897,8 +897,17 @@ update_search_cutoff_1: |-
...
+    "J. R. R.",
+    "W. E. B.",
+  })
+reset_dictionary_1: |-
+  client.Index("books").ResetDictionary()
 get_separator_tokens_1: |-
-  client.Index("articles").GetSeparatorTokens()
+  client.Index("books").GetSeparatorTokens()
```
Comment (curquiza) [inline .code-samples.meilisearch.yaml:910]:
```suggestion
  client.Index("articles").GetSeparatorTokens()
```

**Review by curquiza — CHANGES_REQUESTED** [review]

**Review by curquiza — APPROVED** [review]
niiiice

bors merge

**Pull request merged**